OpenSimRT/RealTime/experimental/RealTimeAnalysisExtended/tests/TestRTExtFromFile.cpp

```diff
@@ -0,0 +1,320 @@
+    auto lHeelCoPLocation =
+            ini.getSimtkVec(section, "LEFT_HEEL_STATION_LOCATION", Vec3(0));
+    auto rToeCoPLocation =
+            ini.getSimtkVec(section, "RIGHT_TOE_STATION_LOCATION", Vec3(0));
+    auto lToeCoPLocation =
+            ini.getSimtkVec(section, "LEFT_TOE_STATION_LOCATION", Vec3(0));
+    auto directionWindowSize =
+            ini.getInteger(section, "DIRECTION_WINDOW_SIZE", 0);
+
+    // so parameters
+    bool solveMuscleOptimization = ini.getBoolean(section, "SOLVE_SO", true);
+    auto convergenceTolerance =
+            ini.getReal(section, "CONVERGENCE_TOLERANCE", 0.0);
+    auto memoryHistory = ini.getInteger(section, "MEMORY_HISTORY", 0);
+    auto maximumIterations = ini.getInteger(section, "MAXIMUM_ITERATIONS", 0);
+    auto objectiveExponent = ini.getInteger(section, "OBJECTIVE_EXPONENT", 0);
+    auto momentArmLibraryPath =
+            ini.getString(section, "MOMENT_ARM_LIBRARY", "");
+
+    // prepare model
+    Object::RegisterType(Thelen2003Muscle());
+    Model model(modelFile);
+    auto state = model.initSystem();
+    const int dofs = state.getNU();
+    const int joints = model.getJointSet().getSize();
+    const int muscles = model.getMuscles().getSize();
+
+    // load moment arm function
+    auto calcMomentArm = OpenSimUtils::getMomentArmFromDynamicLibrary(
+            model, momentArmLibraryPath);
+
+    // prepare marker tasks
+    MarkerData markerData(trcFile);
+    vector<InverseKinematics::MarkerTask> markerTasks;
+    vector<string> observationOrder;
+    InverseKinematics::createMarkerTasksFromMarkerData(
+            model, markerData, markerTasks, observationOrder);
+
+    // read external forces
+    Storage grfMotion(grfMotFile);
+    ExternalWrench::Parameters grfRightFootPar{
+            grfRightApplyBody, grfRightForceExpressed, grfRightPointExpressed};
+    auto grfRightLabels = ExternalWrench::createGRFLabelsFromIdentifiers(
+            grfRightPointIdentifier, grfRightForceIdentifier,
+            grfRightTorqueIdentifier);
+    ExternalWrench::Parameters grfLeftFootPar{
+            grfLeftApplyBody, grfLeftForceExpressed, grfLeftPointExpressed};
+    auto grfLeftLabels = ExternalWrench::createGRFLabelsFromIdentifiers(
+            grfLeftPointIdentifier, grfLeftForceIdentifier,
+            grfLeftTorqueIdentifier);
+    vector<ExternalWrench::Parameters> wrenchParameters;
+    wrenchParameters.push_back(grfRightFootPar);
+    wrenchParameters.push_back(grfLeftFootPar);
+
+    // acquisition function (simulates acquisition from motion)
+    auto dataAcquisitionFunction = [&]() -> MotionCaptureInput {
+        static int i = 0;
+        MotionCaptureInput input;
+
+        // get frame data
+        input.IkFrame = InverseKinematics::getFrameFromMarkerData(
+                i, markerData, observationOrder, false);
+        double t = input.IkFrame.t;
+
+        // dummy delay to simulate real time
+        this_thread::sleep_for(chrono::milliseconds(16));
```

Comment:
Why 16ms and not 10?

Reply:
The dataset is not resampled at 100Hz. The datarate in the .trc file is 60Hz. Although it is not consistent...